crates/biome_js_analyze/src/aria_analyzers/nursery/use_valid_aria_role.rs

```diff
@@ -0,0 +1,216 @@
+use bpaf::Bpaf;
+use serde::{Deserialize, Serialize};
+
+declare_rule! {
+    /// Elements with ARIA roles must use a valid, non-abstract ARIA role.
+    ///
+    /// Source: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/aria-role.md
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="datepicker"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role="range"></div>
+    ///  ```
+    ///
+    ///  ```js,expect_diagnostic
+    ///   <div role=""></div>
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    ///   <Foo role="foo"></Foo>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// <>
+    ///   <div role="button"></div>
+    ///   <div role={role}></div>
+    ///   <div></div>
+    /// </>
+    /// ```
+    ///
+    /// ### Options
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options": {
+    ///         "allowInvalidRoles": ["invalid", "text"],
+    ///         "nonIgnoreDom": true
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ## Accessibility guidelines
+    /// - [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+    ///
+    /// ## Resources
+    /// - [Chrome Audit Rules, AX_ARIA_01](https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_01)
+    /// - [DPUB-ARIA roles](https://www.w3.org/TR/dpub-aria-1.0/)
+    /// - [MDN: Using ARIA: Roles, states, and properties](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques)
+    ///
+    pub(crate) UseValidAriaRole {
+        version: "next",
+        name: "useValidAriaRole",
+        recommended: true,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+#[derive(Deserialize, Serialize, Eq, PartialEq, Debug, Clone, Bpaf)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct ValidAriaRoleOptions {
+    #[bpaf(hide, argument::<String>("roles"), many)]
+    allowed_invalid_roles: Vec<String>,
+    #[bpaf(hide)]
+    ignore_non_dom: bool,
+}
+
+impl Default for ValidAriaRoleOptions {
+    fn default() -> Self {
+        Self {
+            allowed_invalid_roles: Vec::new(),
+            ignore_non_dom: false,
+        }
+    }
+}
+
+impl ValidAriaRoleOptions {
+    pub const KNOWN_KEYS: &'static [&'static str] = &["allowInvalidRoles", "ignoreNonDom"];
+}
+
+impl FromStr for ValidAriaRoleOptions {
+    type Err = ();
+
+    fn from_str(_s: &str) -> Result<Self, Self::Err> {
+        Ok(ValidAriaRoleOptions::default())
+    }
+}
+
+impl VisitNode<JsonLanguage> for ValidAriaRoleOptions {
+    fn visit_member_name(
+        &mut self,
+        node: &SyntaxNode<JsonLanguage>,
+        diagnostics: &mut Vec<DeserializationDiagnostic>,
+    ) -> Option<()> {
+        has_only_known_keys(node, Self::KNOWN_KEYS, diagnostics)
+    }
```

Comment:
Don't forget to rebase your work on main. I pushed some changes related to config parsing. In the process I removed `visit_member_name`. I introduced `report_unknown_map_key` that you can use directly in `visit_map`.